Release 4.2 dev2 and publish release notes (#686)
godotengine/godot-website: collections/_article/dev-snapshot-godot-4-2-dev-2.md

---
title: "Dev snapshot: Godot 4.2 dev 2"

> excerpt: "Closing the first month of development of Godot 4.2, the second development snapshots features both smaller improvements and bigger reworks."

Reviewer: ```suggestion
excerpt: "Closing the first month of development of Godot 4.2, the second development snapshots features both smaller improvements and new features."
```

Author: ```suggestion
excerpt: "Closing the first month of development of Godot 4.2, the second development snapshot includes both smaller improvements and new features."
```

categories: ["pre-release"]
author: Yuri Sizov
image: /storage/blog/covers/dev-snapshot-godot-4-2-dev-2.webp
image_caption_title: "Design visualization system"
image_caption_description: "A VR tool by Dan Silden"
date: 2023-07-28 16:00:00
---

> Final days of July mark the end of the first month of the Godot 4.2 development cycle, and things are starting to pick up steam. With Godot contributors having a few weeks to agree on, work on, review, and approve some of the new features, bigger changes are going to appear in the upcoming dev snapshots.

Reviewer: ```suggestion
These final days of July mark the end of the first month of the Godot 4.2 development cycle, and things are starting to pick up steam. With Godot contributors having had a few weeks to agree on, work on, review, and approve some of the new features, bigger changes are starting to appear in the upcoming dev snapshots.
```

> As always, we try to offer a mix of smaller improvements, bug fixes, and bigger reworks for the Godot community to test and provide feedback. Here are some of the highlights in 4.2 dev2, with a bigger list available [below](#whats-new):

Reviewer: ```suggestion
As always, we try to offer a mix of smaller improvements, bug fixes, and new features for the Godot community to test and provide feedback. Here are some of the highlights in 4.2 dev2, with a bigger list available [below](#whats-new):
```

I suggest changing "bigger reworks" to "new features". They have a different meaning, but both are accurate in this case. The reason I prefer to avoid "bigger reworks" is it may give the impression that we are breaking compatibility by reworking existing features. 

While it is true that we are reworking some existing features, we are trying to do so in a way that maintains compatibility and only improves user experience. 

Since we also have cool new features coming, I think we can just highlight those and avoid the can of worms

Author: I agree with this logic. But I would keep "bigger", as I wanted to strongly imply that we aren't only doing some insignificant tweaks and tune-ups.

```suggestion
As always, we try to offer a mix of smaller improvements, bug fixes, and bigger features for the Godot community to test and provide feedback. Here are some of the highlights in 4.2 dev2, with a bigger list available [below](#whats-new):
```

- The tilemaps system and the tilemaps editor are going through an internal rework, aimed at creating a foundation for the most requested improvements ([GH-78328](https://github.com/godotengine/godot/pull/78328)). The work will continue and you should see more user-facing changes in the upcoming builds. In the meantime a menagerie of workflow enhancements has been added, both in a form of improved tooling ([GH-79512](https://github.com/godotengine/godot/pull/79512), [GH-79899](https://github.com/godotengine/godot/pull/79899)) and more UI hints ([GH-79676](https://github.com/godotengine/godot/pull/79676), [GH-79904](https://github.com/godotengine/godot/pull/79904)).

> - In the core area of the engine maintainers have worked on addressing some design limitations. With these changes included you should no longer experience crashes when trying to change scenes on the fly ([GH-78988](https://github.com/godotengine/godot/pull/78988)) or when renaming nodes during the `ready` step ([GH-78706](https://github.com/godotengine/godot/pull/78706)).

Reviewer: ```suggestion
- In the core area of the engine, maintainers have worked on addressing some design limitations. With these changes included you should no longer experience crashes when trying to change scenes on the fly ([GH-78988](https://github.com/godotengine/godot/pull/78988)) or when renaming nodes during the `ready` step ([GH-78706](https://github.com/godotengine/godot/pull/78706)).
```

> - You can now interact with and extend the OpenXR API through GDExtension ([GH-68259](https://github.com/godotengine/godot/pull/68259))! Support for another previously missing feature, indexed properties, have also been added ([GH-79763](https://github.com/godotengine/godot/pull/79763)).

Reviewer: ```suggestion
- You can now interact with and extend the OpenXR API through GDExtension ([GH-68259](https://github.com/godotengine/godot/pull/68259))! Support for another previously missing feature, indexed properties, has also been added ([GH-79763](https://github.com/godotengine/godot/pull/79763)).
```
